packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -37,13 +37,13 @@ enum _ScaffoldSlot {
 
 class _ScaffoldLayout extends MultiChildLayoutDelegate {
   _ScaffoldLayout({
-    @required this.padding,
     @required this.statusBarHeight,
+    @required this.bottomPadding,
     @required this.textDirection,
   });
 
-  final EdgeInsets padding;
   final double statusBarHeight;
+  final double bottomPadding;
```

Comment:
Curious: what situation is this meant to handle? While this is consistent with the current code, this actually results in undesirable behaviour on the iPhone X where the additional padding shouldn't be applied at the Scaffold level, but at the BottomNavigationBar level. I assume the current behaviour is desirable in the case it's meant to cover though.

I ran into this while implementing iPhone X bottom nav bar padding.

Reply:
Actually reading further, this looks like this may well my issue, but still curious what the origin of this was.

Reply:
The `padding` -> `bottomPadding` part specifically should have been a no-op. As far as I can tell all the other sides of the padding were ignored in the old code.

Reply:
Mostly interested in the rationale for the existing behaviour. Though probably good to deal with that in a followup patch when we've got the changes to BottomNavigationBar etc. to support the iPhone X.

Reply:
I didn't really examine what the existing behaviour actually was. :-) Scaffold could probably do with a round of cleanup, it's accumulated cruft over the years.